### PR TITLE
Esdt-Safe update + cleanup

### PR DESCRIPTION
**Pull request by andreiblt1304**

### CODE REVIEW

File: esdt-safe/src/from_sovereign/transfer_tokens.rs [hunk 1]
```diff
@@ -138,29 +142,26 @@ pub trait TransferTokensModule:
...
 
                 self.tx()
                     .to(&operation_tuple.operation.to)
                     .raw_call(transfer_data.function.clone())
-                    .arguments_raw(args.clone())
+                    .arguments_raw(args)
                     .multi_esdt(mapped_tokens.clone())
                     .gas(transfer_data.gas_limit)
                     .callback(
-                        <Self as TransferTokensModule>::callbacks(self)
```
Comment (andreiblt1304) [inline esdt-safe/src/from_sovereign/transfer_tokens.rs:162]: this will be reverted

File: esdt-safe/src/to_sovereign/create_tx.rs [hunk 2]
```diff
@@ -69,40 +65,33 @@ pub trait CreateTxModule:
 
             current_token_data.amount = payment.amount.clone();
 
-            if self.is_sovereign_chain().get() {
+            let is_sovereign_chain = self.is_sovereign_chain().get();
+
```
Comment (axenteoctavian) [inline esdt-safe/src/to_sovereign/create_tx.rs:69]: You can delete this empty line
```suggestion
```

File: esdt-safe/src/to_sovereign/create_tx.rs [hunk 3]
```diff
@@ -69,40 +65,33 @@ pub trait CreateTxModule:
...
                     .transfer_execute();
 
-                event_payments.push(MultiValue3((
+                let event_payment: EventPaymentTuple<Self::Api> = MultiValue3((
                     payment.token_identifier.clone(),
                     payment.token_nonce,
                     current_token_data.clone(),
-                )));
+                ));
+
```
Comment (axenteoctavian) [inline esdt-safe/src/to_sovereign/create_tx.rs:82]: You can delete this empty line
```suggestion
```

File: esdt-safe/src/to_sovereign/create_tx.rs [hunk 4]
```diff
@@ -69,40 +65,33 @@ pub trait CreateTxModule:
...
-                        current_token_data.clone(),
-                    )));
-
-                    continue;
-                }
+                let sov_token_nonce = self.burn_mainchain_token(payment, &sov_token_id);
 
-                let sov_token_nonce = self.remove_sovereign_token(payment, &sov_token_id);
+                let event_payment: EventPaymentTuple<Self::Api> =
+                    MultiValue3((sov_token_id, sov_token_nonce, current_token_data.clone()));
```
Comment (axenteoctavian) [inline esdt-safe/src/to_sovereign/create_tx.rs:93]:
```suggestion
```

File: esdt-safe/src/from_sovereign/transfer_tokens.rs [hunk 5]
```diff
@@ -193,28 +195,21 @@ pub trait TransferTokensModule:
...
-            args.push_arg(token.amount);
-        }
+        args.push_multi_arg(&mapped_tokens);
 
         args
     }
 
     #[promises_callback]
-    fn execute(
+    fn distribute_payments_callback(
```
Comment (axenteoctavian) [inline esdt-safe/src/from_sovereign/transfer_tokens.rs:204]: From what I remember this function name is actually the Identifier for the events in the function and we explicitly want `execute` the identifier. Please revert this change.
```suggestion
    fn execute(
```

**Review by axenteoctavian — CHANGES_REQUESTED** [review]

**Review by axenteoctavian — APPROVED** [review]

**Pull request merged**